### PR TITLE
utils_test.__init__: Correct time format error

### PR DESCRIPTION
**Pull request by sitoliu**
id: 1490280
Signed-off-by: Sitong Liu siliu@redhat.com

### CODE REVIEW

**Comment by sitoliu** [comment]
smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_scsi.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.ntp.with_reboot: PASS (605.66 s)

**Comment by yanan-fu** [comment]
@sitoliu  It is better to add more detailed explanation for this change in commit message.

**Comment by vivianQizhu** [comment]
Acked-by: Qianqian Zhu qizhu@redhat.com

**Comment by vivianQizhu** [comment]
Commit title format: 1. Add module name before description. 2. Title length should not exceed 70 char. 3. Signature please.

**Comment by yanan-fu** [comment]
please update signature again !

**Comment by yanan-fu** [comment]
LGTM
Acked-by: Yanan Fu <yfu@redhat.com>

**Comment by sitoliu** [comment]
Thanks all, have updated the commit info.

**Comment by luckyh** [comment]
Good catch, so we need to deal with `12:00:00 AM` too.

**Comment by sitoliu** [comment]
Thanks @luckyh , have updated the code.

File: virttest/utils_test/__init__.py [hunk 1]
```diff
@@ -242,11 +242,16 @@ def get_time(session, time_command, time_filter_re, time_format):
...
             if re.search('PM', guest_time):
                 hour = re.findall('\d+ (\d+):', guest_time)[0]
-                hour = str(int(hour) + 12)
+                if int(hour) < 12:
+                    hour = str(int(hour) + 12)
                 guest_time = re.sub('\s\d+:', " %s:" % hour,
                                     guest_time)[:-3]
             else:
-                guest_time = guest_time[:-3]
+                hour = re.findall('\d+ (\d+):', guest_time)[0]
```
Comment (yanan-fu) [inline virttest/utils_test/__init__.py]: `else` here include 24 hour format  and 12 hour format with "AM".  and this change only suitable for the second scenario.

Comment (yanan-fu) [reply virttest/utils_test/__init__.py]: `hour = re.findall('\d+ (\d+):', guest_time)[0]`  we move it out of if-else, as it is used not only once.

File: virttest/utils_test/__init__.py [hunk 2]
```diff
@@ -242,7 +242,14 @@ def get_time(session, time_command, time_filter_re, time_format):
...
-                hour = str(int(hour) + 12)
+                if int(hour) < 12:
+                    hour = str(int(hour) + 12)
+                guest_time = re.sub('\s\d+:', " %s:" % hour,
+                                    guest_time)[:-3]
+            elif re.search('AM', guest_time):
+                hour = re.findall('\d+ (\d+):', guest_time)[0]
+                if int(hour) == 12:
+                    hour = str(int(hour) - 12)
                 guest_time = re.sub('\s\d+:', " %s:" % hour,
```
Comment (yanan-fu) [inline virttest/utils_test/__init__.py]: L253 and L250 are duplicate with L247 and L244,  how about replace L243~L256 with: 

`hour = re.findall('\d+ (\d+):', guest_time)[0]           
 if re.search('PM', guest_time):
     if int(hour) < 12:
         hour = str(int(hour) + 12)
elif re.search('AM', guest_time):
    if int(hour) == 12:
        hour = str(int(hour) - 12)
guest_time = re.sub('\s\d+:', " %s:" % hour, guest_time)[:-3]`

**Comment by sitoliu** [comment]
Thanks @yanan-fu , updated done, and will add the result later.

**Comment by sitoliu** [comment]
New test passed.
smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_scsi.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.ntp.with_reboot: PASS (483.96 s)

File: virttest/utils_test/__init__.py [hunk 3]
```diff
@@ -240,9 +240,14 @@ def get_time(session, time_command, time_filter_re, time_format):
...
-            if re.search('PM', guest_time):
+            if re.search('PM|AM', guest_time):
                 hour = re.findall('\d+ (\d+):', guest_time)[0]
-                hour = str(int(hour) + 12)
+                if re.search('PM', guest_time):
+                    if int(hour) < 12:
+                        hour = str(int(hour) + 12)
+                else:
+                    if int(hour) == 12:
+                        hour = str(int(hour) - 12)
```
Comment (luckyh) [inline virttest/utils_test/__init__.py]: I prefer the following approach:
```python
fix = 12 if re.search('PM', guest_time) else 0
hour = hour % 12 + fix
```

Comment (yanan-fu) [reply virttest/utils_test/__init__.py]: Good method,  cover both  PM  and AM. 👍

Comment (sitoliu) [reply virttest/utils_test/__init__.py]: Cool! Will update it. Thanks!

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**